runtime/drivers/file/watcher.go

```diff
@@ -193,6 +194,9 @@ func (w *watcher) runInner() error {
 func (w *watcher) addDir(path string, replay bool) error {
 	err := w.watcher.Add(path)
 	if err != nil {
+		if os.IsNotExist(err) || errors.Is(err, unix.ENOENT) {
```

Comment:
Does the upstream code handle cases when its a genuine error that file being watched doesn't exist ? 
Could it be problematic to not return an error if trying to watch a non existent directory ?

Reply:
No it doesn't for the root directory case, good catch! Should be fixed now :)